tests/spread/hooks/task.yaml

```diff
@@ -0,0 +1,32 @@
+summary: test snap hook of charmcraft
+
+systems:
+  - ubuntu-22.04-64
+
+include: 
+  - tests/
+
+prepare: |
+  tests.pkgs install unzip
+  charmcraft init --project-dir=charm
+
+restore: |
+  if stat /charmcraft/charmcraft_*.snap 2>/dev/null; then
```

Comment:
Rather than duplicating code, can you make this a function and call the function here and in the main `spread.yaml`?